scrubadub/scrubbers.py

```diff
@@ -222,7 +222,8 @@ def _post_process_filth_list(self, filth_list: Sequence[Filth]) -> Sequence[Filt
         return filth_list
 
     def iter_filth(
-            self, text: str, document_name: Optional[str] = None, run_post_processors: bool = True
+            self, text: str, document_name: Optional[str] = None, run_post_processors: bool = True,
+            run_merge: bool = True, exclude_detectors: Optional[List[str]] = None
```

Comment:
The scrubber is looking a little complex now. What if we could simplify it a little? How about making `iter_filth` only call `iter_filth_documents` and we do all the heavy lifting in there?